pkg/ui/common/helpers.go

```diff
@@ -23,16 +29,33 @@ type onlyRootFS struct {
 	path string
 }
 
+// NewOnlyRootFS returns a new FS that only allows access to the root directory.
 func NewOnlyRootFS(fs fs.FS, path string) fs.FS {
 	return &onlyRootFS{
 		fs,
 		path,
 	}
 }
 
+// Open opens the named file.
 func (f onlyRootFS) Open(name string) (fs.File, error) {
 	if name != "." {
 		return nil, &fs.PathError{Op: "open", Path: name, Err: fs.ErrNotExist}
 	}
 	return f.FS.Open(f.path)
 }
+
+// ErrorMessageForUI returns the error message of an error, rewritten for simplicity in the UI.
+func ErrorMessageForUI(field, errMsg string) string {
```

Comment:
why do we need such complexity? can't we just return needed error message right from the `service`, `controller` etc?

Reply:
these messages look really bad in the UI